OutOfSchool/OutOfSchool.WebApi/Services/ChildService.cs

```diff
@@ -27,112 +28,77 @@ public ChildService(IEntityRepository<Child> repository)
-            }
+            var children = await repository.GetAll().ConfigureAwait(false);
+
+            return Result<IEnumerable<ChildDTO>>.GetSuccess(children.Select(child => child.ToModel()).ToList());
         }
 
         /// <inheritdoc/>
-        public async Task<ChildDTO> GetById(long id)
+        public async Task<Result<ChildDTO>> GetById(long id)
         {
             var child = await repository.GetById(id).ConfigureAwait(false);
 
             if (child == null)
             {
-                throw new ArgumentException("Incorrect Id!", nameof(id));
+                return Result<ChildDTO>.GetError(ErrorCode.NotFound, $"Child with id = {id} was not found.");
             }
 
-            return child.ToModel();
+            return Result<ChildDTO>.GetSuccess(child.ToModel());
         }
 
         /// <inheritdoc/>
-        public async Task<ChildDTO> Update(ChildDTO dto)
+        public async Task<Result<ChildDTO>> Update(ChildDTO dto)
         {
-            if (dto == null)
-            {
-                throw new ArgumentNullException(nameof(dto), "Child was null.");
-            }
-
-            if (dto.DateOfBirth > DateTime.Now)
-            {
-                throw new ArgumentException("Wrong date of birth.", nameof(dto));
-            }
-
-            if (dto.FirstName.Length == 0)
-            {
-                throw new ArgumentException("Empty firstname.", nameof(dto));
-            }
-
-            if (dto.LastName.Length == 0)
-            {
-                throw new ArgumentException("Empty lastname.", nameof(dto));
-            }
-
-            if (dto.MiddleName.Length == 0)
-            {
-                throw new ArgumentException("Empty middlename.", nameof(dto));
-            }
-
             try
             {
+                if (dto == null)
+                {
+                    return Result<ChildDTO>.GetError(ErrorCode.NotFound, "Child item is null.");
+                }
+
                 var child = await repository.Update(dto.ToDomain()).ConfigureAwait(false);
-              
-                return child.ToModel();
+
+                return Result<ChildDTO>.GetSuccess(child.ToModel());
             }
-            catch (Exception ex)
+            catch
             {
-                throw new Exception($"{nameof(dto)} could not be updated: {ex.Message}");
+                return Result<ChildDTO>.GetError(ErrorCode.InternalServerError, "Internal server error.");
             }
         }
 
         /// <inheritdoc/>
-        public async Task Delete(long id)
+        public async Task<Result<long>> Delete(long id)
         {
             try
             {
-                await repository
-                    .Delete(await repository.GetById(id).ConfigureAwait(false))
-                    .ConfigureAwait(false);
+                var dtoToDelete = new ChildDTO() { Id = id };
+                
+                await repository.Delete(dtoToDelete.ToDomain()).ConfigureAwait(false);
```

Comment:
could we have a Delete method that takes just an id as a parameter?

Reply:
Do I need to change the implementation of Repository? If yes, I suppose it should be only in this way as shown on the screenshot.
![image](https://user-images.githubusercontent.com/55458556/109637541-978d7980-7b55-11eb-8287-6fb3f8df80a7.png)

Reply:
then rest of the team must change it as well, and rewrite tests